foreshadow/tests/test_foreshadow.py

```diff
@@ -501,3 +501,26 @@ def test_foreshadow_param_optimize_invalid_dict_key():
         _param_mapping(fs.pipeline, x_train, y_train)
 
     assert str(e.value) == "Invalid JSON Key"
+
+
+def test_core_foreshadow_example():
```

Comment:
Can we also add a test case for classification problem? Better if there is some categorical features as well.

Reply:
This is a good suggestion, I have implemented it in the updated pull request that also merges some changes that Alexander had made.